apps/prairielearn/src/sync/fromDisk/topics.js

```diff
@@ -27,7 +27,11 @@ module.exports.sync = async function (courseId, courseData) {
   /** @type Set<string> */
   const knownQuestionTopicNames = new Set();
   Object.values(courseData.questions).forEach((q) => {
-    if (!infofile.hasErrors(q) && q.data?.topic) {
+    // We technically allow courses to define an "empty string" topic, so we'll
+    // support that for implicit topics as well.
+    //
+    // TODO: consider requiring that all topics have a non-empty name.
+    if (!infofile.hasErrors(q) && typeof q.data?.topic === 'string') {
```

Comment:
Any reason why this is done by checking the type instead of the value?

Reply:
No reason in particular; `!= null` would achieve the same thing and might be clearer? I can change that.

Reply:
Indeed it's clearer. That said, I'm not sure if the test is actually needed, since if it's null it would likely have errors, but I think an explicit check for null here makes the intention clear.

Reply:
That's a fair point about it maybe not actually being needed... but yeah, I'll leave it in for clarify. Done in 44244f5.